cmd/virt-handler/virt-handler.go

```diff
@@ -68,19 +71,41 @@ func main() {
 		panic(err)
 	}
 
+	// Create event recorder
+	coreClient, err := kubecli.Get()
+	if err != nil {
+		panic(err)
+	}
+	broadcaster := record.NewBroadcaster()
+	broadcaster.StartRecordingToSink(&kubecorev1.EventSinkImpl{Interface: coreClient.Events(api.NamespaceDefault)})
+	recorder := broadcaster.NewRecorder(kubev1.EventSource{Component: "virt-handler", Host: *host})
+
 	domainListWatcher := virtcache.NewListWatchFromClient(domainLWConn, libvirtapi.VIR_DOMAIN_EVENT_ID_LIFECYCLE)
 
 	_, domainController := kubecli.NewInformer(domainListWatcher, &libvirt.Domain{}, 0, kubecli.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) error {
-			fmt.Printf("Domain ADDED: %s: %s\n", obj.(*libvirt.Domain).GetObjectMeta().GetName(), obj.(*libvirt.Domain).Status.Status)
+			domain := obj.(*libvirt.Domain)
+			// TODO this just demonstrates how to do event logging. More fine grained differentiations are necessary here.
+			// TODO For now we use an empty reference for event logging, normally whenever an event is logged, an update on the VM object itself is also part of this
+			if domain.Status.Status == libvirt.Running {
+				recorder.Event(NewVMReferenceFromDomain(domain), kubev1.EventTypeNormal, "Started", "VM started")
```

Comment:
I'm not so much worried about the load on the API server. I'm rather more concerned about the semantics of the events.
To me the question is what kind of event do we want to spawn when?

Currently it looks like the events are rather coupled to th evirt-handler state. This is okay if we want this semantics.